tests/test_cms_menus.py

```diff
@@ -404,6 +405,239 @@ def test_menu_without_softroots(self):
+
+        page = self.bbb_pagecontent.page
+        context = self.get_context(page.get_absolute_url(), page=page)
+        tpl = Template("{% load menu_tags %}{% show_menu 0 100 100 100 %}")
+        tpl.render(context)
+        hard_root = context['children']
+
+        mock_tree = [
+            AttributeObject(title=root.title, level=0, children=[
+                AttributeObject(title=bbb.title, level=1, children=[])
+            ])
+        ]
+
+        self.assertTreeQuality(hard_root, mock_tree, 'level', 'title')
+
+    def test_menu_with_softroot_and_hidden_child(self):
+        """
+        Tree in fixture :
+               root
+                   aaa( soft_root is True )
+                       aaa1 ( in_navigaton is False)
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 0 100
+        expected result 1:
+                0:root
+                  1:aaa
+                  1:bbb
+        Note: If node has in_navigation False , node and its descendants are removed from tree
+        """
+        menu_content = factories.MenuContentWithVersionFactory(version__state=PUBLISHED, language=self.language)
+        root = factories.ChildMenuItemFactory(parent=menu_content.root, content=self.root_pagecontent.page)
+        aaa = factories.ChildMenuItemFactory(parent=root, content=self.aaa_pagecontent.page, soft_root=True)
+        aaa1 = factories.ChildMenuItemFactory(parent=aaa, content=self.aaa1_pagecontent.page)
+        ccc = factories.ChildMenuItemFactory(parent=aaa1, content=self.ccc_pagecontent.page)
+        factories.ChildMenuItemFactory(parent=ccc, content=self.ddd_pagecontent.page)
+        factories.ChildMenuItemFactory(parent=aaa, content=self.aaa2_pagecontent.page, hide_node=True)
+        bbb = factories.ChildMenuItemFactory(parent=root, content=self.bbb_pagecontent.page)
+
+        page = self.bbb_pagecontent.page
+        context = self.get_context(page.get_absolute_url(), page=page)
+        tpl = Template("{% load menu_tags %}{% show_menu 0 100 100 100 %}")
+        tpl.render(context)
+        hard_root = context['children']
+
+        mock_tree = [
+            AttributeObject(title=root.title, level=0, children=[
+                AttributeObject(title=aaa.title, level=1, children=[]),
+                AttributeObject(title=bbb.title, level=1, children=[])
+            ])
+        ]
+
+        self.assertTreeQuality(hard_root, mock_tree, 'level', 'title')
+
+    def test_menu_with_rendering_hidden_softroot(self):
+        """
+        Tree in fixture :
+               root
+                   aaa( soft_root is True and hide_noe is True)
+                       aaa1
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 0 100
+        expected result 1:
```

Comment:
I don't understand why this is the expected result. Please add more detail to the description to help me understand what we are trying to test here. If I had to repair this in the future I would have no idea.